sparta/test/Pipeline/Pipeline_test.cpp

```diff
@@ -288,6 +292,8 @@ int main ()
 
     sparta::Pipeline<uint64_t> examplePipeline7("mySeventhSpartaPipeline", 2, root_clk.get());
 
+    sparta::Pipeline<uint64_t, sparta::PhasedPayloadEvent<uint64_t>> examplePipeline8(&es, "myEighthSpartaPipeline", 3, root_clk.get());
```

Comment:
Curious, did you attempt to pass and event type whose template argument != DataT?  What would happen here?
```
sparta::Pipeline<uint64_t, sparta::PhasedPayloadEvent<std::string>>
```

Reply:
I didn't try it. But I think it probably work until the handler is actually called by Pipeline. In you example, PayloadEvent is expecting std::string but Pipeline feeds DataT ...

Eventually I have an assertion to make sure PhasedPayloadEvent uses the template argument DataT.

Reply:
Actually, I just tried it -- it successfully fails with the new static_assert you just added.  😁 
```
.././sparta/resources/Pipeline.hpp:253:13: error: static_assert failed due to requirement 'std::is_same_v<sparta::PhasedPayloadEvent<int>, sparta::PhasedUniqueEvent> || std::is_same_v<sparta::PhasedPayloadEvent<int>, sparta::PhasedPayloadEvent<unsigned long> >' "Pipeline is templated a unsupported Event type. Supported Event types: {PhasedUniqueEvent, PhasedPayloaodEvent<DataT>}"
```
when I did this:
```
sparta::Pipeline<uint64_t, sparta::PhasedPayloadEvent<int>> examplePipeline8(&es, "myEighthSpartaPipeline", 3, root_clk.get());
```

Reply:
However, the error message should look more like this:
"Error: Pipeline is templated on a unsupported Event type. Supported Event types: UniqueEvent, PayloaodEvent<DataT> (where DataT == DataT of the Pipeline). "